src/WebJobs.Script/Config/ScriptSettingsManager.cs

```diff
@@ -68,6 +68,31 @@ public virtual string AzureWebsiteUniqueSlotName
             }
         }
 
+        public virtual string ApplicationInsightsInstrumentationKey
+        {
+            get => GetSettingFromCache(EnvironmentSettingNames.AppInsightsInstrumentationKey);
+            set => UpdateSettingInCache(EnvironmentSettingNames.AppInsightsInstrumentationKey, value);
+        }
+
+        private string GetSettingFromCache(string settingKey)
+        {
+            string settingValue = null;
+            if (!string.IsNullOrEmpty(settingKey))
```

Comment:
Should we throw here instead? Are there any scenarios where it would be valid to call this with a `null` key?